front/app/components/FilterBoxes/TopicsFilter.tsx

```diff
@@ -140,48 +137,28 @@ const TopicsFilter = memo<Props>(
         >
           <Box>
             <Box className="e2e-topics-filters" aria-live="polite">
-              {topicsWithIdeas
-                .filter((topic) => !isError(topic))
-                .map((topic: ITopicData, index) => {
-                  const postCount = get(
-                    filterCounts,
-                    `topic_id.${topic.id}`,
-                    0
-                  );
-
-                  const topicSelected = selectedTopicIds?.includes(topic.id);
-
-                  if (!showFullList && index > 4) {
-                    // Only show the first 5 topics by default
-                    return null;
-                  }
```

Comment:
This code could perhaps confuse us in the future? :) And we'd be doing array operations over the full array still. I've changed it to a sliced version if the full list is not shown.

Reply:
Good call! :+1: